### PR TITLE
Add a `waypoint fmt` command

### LINKED ISSUE #705: Have a waypoint fmt to format the waypoint.hcl
**Is your feature request related to a problem? Please describe.**

With a waypoint init, I get an extra set of whitespace.

**Describe the solution you'd like**

I would like to have `waypoint fmt` just like a `terraform fmt` command that format the HCL content

**Describe alternatives you've considered**

Using `terraform fmt`

### PR DESCRIPTION
**Pull request by mitchellh**
Fixes #705 

This adds a `waypoint fmt` command to format `waypoint.hcl` files. This is a simplified version of `terraform fmt`: https://www.terraform.io/docs/cli/commands/fmt.html and the underlying logic is identical.

### CODE REVIEW

File: internal/cli/fmt.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+func (c *FmtCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(sets *flag.Sets) {
+		f := sets.NewSet("Command Options")
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "write",
+			Target:  &c.flagWrite,
+			Default: false,
+			Usage: "Overwrite the input file. If this is false, the formatted " +
```
Comment (briancain) [inline internal/cli/fmt.go:106]: Quick note, this should be reversed to have the opposite behavior. Always write in place, and can be disabled and write to stdout with `-write=false`

**Review by briancain — CHANGES_REQUESTED** [review]
Works well for me! Just a quick note about the `-write` behavior to match `terraform fmt`.

**Review by briancain — APPROVED** [review]

File: internal/cli/fmt.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		WithFlags(c.Flags()),
+		WithNoConfig(),
+		WithClient(false),
+	); err != nil {
+		return 1
+	}
+
+	// If we have too many args, error immediately.
+	if len(c.args) > 1 {
+		c.ui.Output("At most one argument is expected.\n\n"+c.Help(), terminal.WithErrorStyle())
```
Comment (briancain) [inline internal/cli/fmt.go:36]: I noticed this should maaybe be increased if FILE is provided? So if you want to format a waypoint.hcl in a different  folder but to STDOUT with `-write=false`:

```
brian@localghost:docker % waypoint fmt angular/waypoint.hcl -write=false                                                                                                                                                                ±[●][main]
! At most one argument is expected.
```

Comment (mitchellh) [reply internal/cli/fmt.go:36]: Don't think this is a bug, flags in Go have to be before args, so `-write=false angular/waypoint.hcl` should work.

**Pull request merged**